packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,650 @@
+
+    Conversely, this event will not be fired when:
+
+    1. A user moves the cursor away either using mouse or any key down events that moves the cursor from current position to outside of `@`
+       and characters after that with less than two white spaces while the popup is open.
+    2. A user selects an option from the dropdown list either using mouse click/pressing Enter or Tab.
+    3. A user adds two or more spaces after triggering the mention popup.
+    4. In the absence of `nimble-rich-text-mention-users` and no mapping elements.
+
+#### Other APIs for `@mention`
+
+To support the `@mention`, the creation of the following configuration elements and UI elements is required for rich text components:
+
+#### User mention element (Non-visible configuration element):
+
+This element serves as a container for the mapping elements and additionally functions as a configuration element. It allows the transmission of
+arbitrary values that are specific to the `@mention` users within the rich text components. Currently, this component does not have any
+configuration properties/attributes for user mentions and plays the role of containing element for the user mapping elements.
+
+This component can be used as a feature flag to enable the support for `@mention` in the rich text editor and viewer.
+If this component is not present at the children of the rich text editor or viewer, the component will function the same without the support
+for `@mention` in any place like editor, viewer, markdown parser and markdown serializer and the toolbar button to add **"@"** in the editor
+will be hidden.
+
+_Component Name_
+
+-   `nimble-rich-text-mention-users`
+
+_Content_
+
+-   Zero or more `nimble-mapping-mention-user` elements
+
+#### User mapping element (Non-visible configuration element):
+
+This mapping element is employed to establish a connection between the value displayed in the mapping view and the corresponding value stored within
+the markdown string. For instance, the username for an `@mention` is contained in the `display-name` attribute, which is used for display in the
+mention view, while the user ID URL is contained in the `mention-url` attribute which is used to store within the markdown string.
+
+_Component Name_
+
+-   `nimble-mapping-mention-user`
+
+_Props/Attrs_
+
+-   `mention-url`: string
+-   `display-name`: string
+
+#### User mention view (Visible UI element):
+
+This is a UI component used to render the `@mention` node in rich text editor and rich text viewer when parsed or added into the components.
+This holds the styling for the `@mention` nodes.
+
+_Component Name_
+
+-   `nimble-rich-text-user-mention-view`
+
+_Props/Attrs_
+
+-   `mention-url`: string - the user ID URL of the mentioned user
+-   `mention-label`: string - the user name of the mentioned user
```

Comment:
@rajsite and I realized there is a corner case to consider: what should render if the markdown contains an @mention with the proper URL syntax but no corresponding user mention element? This could happen if:
1. a user was @mentioned but then they left the company so they are no longer a valid user
2. the client is loading user info asynchronously and hasn't populated the user mapping for all mentioned users yet, but will do so soon
3. someone pastes an absolute URL that matches the regex for @mention but isn't actually a valid user ID

For 1 and 2 I could imagine us rendering something that looks like an @mention but shows the user ID or "not found".

For 3 we might want to render it as the raw absolute URL.

But I don't think the rich text components can tell whether it's case 1, 2, or 3 so we probably have to pick one of those two rendering options.

Reply:
For 1 and 2, we will render as `user ID` fetched from the URL. This is applicable only when parsing the markdown string to a mentioned node. So the label and content of the node will have a user ID in place of the user name and render it as a mentioned node.

For 3, by default, it renders as an absolute link if it doesn't match the regex pattern as it detects the URL is not part of a mention node, it will be considered as an `autolink` as it gets the next precedence in the markdown parser.